Document MWT and Widgets debug utilities. (#308)
MicroEJ/docs: ApplicationDeveloperGuide/UI/Widgets/index.rst

- ``OverlapContainer``: lays out any number of children by stacking them.
- ``Canvas``: lays out any number of children freely.

Debug Utilities
~~~~~~~~~~~~~~~

A few utilities useful for debugging are available in the package ``ej.widget.util.debug`` of the widget library.

> - Print the hierarchy of widgets.

Reviewer: use subsection rather than bullet?

Author: Yes, I did that to avoid going too deep in the section levels but maybe it's clearer.

Reviewer 2: would be nice, what abut having each section of "Widgets and examples" in separated pages for this section levels issue?

  The method ``HierarchyInspector.hierarchyToString(Widget)`` returns a String representing the hierarchy of a widget.
  In other words, it prints the widget and its children recursively in a tree format.

  It may be used to analyse the content of a page and have a quick estimation of the number of widgets and the depth of the hierarchy.

  For example:

  .. code-block::

     Scroll
     +--ScrollableList
     |  +--Label
     |  +--Dock
     |  |  +--ImageWidget
     |  |  +--Label
     |  +--Label
  
- Print the path to a widget.

>   The method ``HierarchyInspector. pathToWidgetToString(Widget)`` returns a String with the hierarchy of containers containing the widget.

Reviewer: it's not really a hierarchy but rather a list i think.
i suggest `returns a String representing the list of ancestors of the widget`

  For example: ``Desktop > Scroll > ScrollableList > Label``.

  It may be used to identify a widget in a trace.

  It is also possible to choose the separator by using ``HierarchyInspector.pathToWidgetToString(Widget, char)`` method.
  For example: ``Desktop ; Scroll ; ScrollableList ; Label``.

- Count the number of widgets or containers.

  The methods ``HierarchyInspector.countNumberOfWidgets(Widget)`` and ``HierarchyInspector.countNumberOfContainers(Widget)`` respectively count the number of widgets and containers in a hierarchy.

>   It may be used to evaluate the complexity of a hierarchy of widget.

Reviewer: widget**s**

- Count the maximum depth of a hierarchy.

  The method ``HierarchyInspector.countMaxDepth(Widget)`` counts the maximum depth of a hierarchy.
  In other words, the depth of the widget with the biggest number of parents recursively.

>   It may be used to evaluate the complexity of a hierarchy of widget.

Reviewer: widget**s**

- Print the bounds of a widget.

>   The method ``BoundsInspector.boundsToString(Widget)`` returns a String with the type of the widget and its bounds.

Reviewer 2: ```suggestion
  The method ``BoundsInspector.boundsToString(Widget)`` returns a String with the widget type, position and size.
```

Author: I didn't want to detail everything because it is explained just after.

>   The String contains:

Reviewer 2: ```suggestion
  The returned String contains:
```

  - the simple name of the class of the widget,
  - its position relative to its parent,
  - its size,
  - its absolute position.

  For example: ``Label: 0,0 7x25 (absolute: 75,75)``

- Print the bounds of all the widgets in a hierarchy.

>   The method ``BoundsInspector.boundsRecursiveToString(Widget)`` returns a String representing the hierarchy of a widget.
>   And for each container or widget, its bounds.

Reviewer 2: ```suggestion
  Similarly, the method ``BoundsInspector.boundsRecursiveToString(Widget)`` returns a String representing the type, position and size of a widget and its ancestors.
```
